sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -221,8 +221,7 @@ object ScalaNativePluginInternal {
             val isCppSource = path.endsWith(".cpp")
 
             val compiler = abs(if (isCppSource) clangpp else clang)
-            val flags = Seq("-O2") ++ (if (isCppSource) Seq("-std=c++11")
-                                       else Seq()) ++ opts
+            val flags    = (if (isCppSource) Seq("-std=c++11") else Seq()) ++ opts
```

Comment:
In my experience `-O2` makes debugging harder and I'd expect that we want to ensure that the runtime library code itself is debug friendly when the user requests it, especially since it in some cases uses C wrappers.

If you insist that this is not the case, I think there is still a bug in that the optimization flag needs to be moved after the common flags to take effect.